contracts/RewardsPoolFactory.sol

```diff
@@ -0,0 +1,167 @@
+     */
+    function deploy(
+        address            _stakingToken,
+        address[] calldata _rewardsTokens,
+        uint256[] calldata _rewardsAmounts,
+        uint256            _rewardsDuration
+    ) external onlyOwner {
+        require(_rewardsDuration != 0, 'RewardsPoolFactory::deploy:The Duration should be greater than zero');
+        require(_rewardsTokens.length != 0, 'RewardsPoolFactory::deploy: RewardsTokens and RewardsAmounts arrays could not be empty');
+        require(_rewardsTokens.length == _rewardsAmounts.length, 'RewardsPoolFactory::deploy: RewardsTokens and RewardsAmounts should have a matching sizes');
+
+        for (uint256 i = 0; i < _rewardsTokens.length; i++) {
+            require(_rewardsTokens[i] != address(0), 'RewardsPoolFactory::deploy: Reward token address could not be invalid');
+            require(_rewardsAmounts[i] != 0, 'RewardsPoolFactory::deploy: Reward must be greater than zero');
+        }
+
+        address rewardsPool = address(new RewardsPool(_rewardsTokens, _rewardsAmounts, _stakingToken, _rewardsDuration));
+
+        rewardsPools.push(rewardsPool);
+
+        emit LiquidityMiningCampaignDeployed(rewardsPool, _stakingToken);
+    }
+
+    /** @dev Function that will extend the rewards period, but not change the reward rate, for a given staking contract.
+     * @param rewardsPoolAddress The address of liquidity mining campaign
+     * @param extendRewardToken The address of the token the rewards will be paid in
+     * @param extendRewardAmount The additional reward amount
+     */
+    function extendRewardPeriod(
+        address rewardsPoolAddress,
+        address extendRewardToken,
+        uint256 extendRewardAmount
+    )
+        external
+        onlyOwner
+    {
+        require(extendRewardAmount != 0, 'RewardsPoolFactory::extendRewardPeriod: Reward must be greater than zero');
+
+        require(rewardsPoolAddress != address(0), 'RewardsPoolFactory::extendRewardPeriod: not deployed');
+        require(hasStakingStarted(rewardsPoolAddress), 'RewardsPoolFactory::extendRewardPeriod: Staking has not started');
+
+        (uint256 rate, , , ,) = RewardsPool(rewardsPoolAddress).rewardsTokensMap(extendRewardToken);
+
+        require(rate != 0, 'RewardsPoolFactory::extendRewardPeriod: Token for extending reward is not known'); // its expected that valid token should have a valid rate
+
+        IERC20Detailed(extendRewardToken).safeApprove(rewardsPoolAddress, extendRewardAmount);
+        RewardsPool(rewardsPoolAddress).addRewards(extendRewardToken, extendRewardAmount);
+    }
+
+    /* ========== Permissionless FUNCTIONS ========== */
+
+    /** @dev Calls startStakings for all staking tokens.
+     */
+    function startStakings() external {
+        require(rewardsPools.length != 0, 'RewardsPoolFactory::startStakings: called before any deploys');
+
+        for (uint256 i = 0; i < rewardsPools.length; i++) {
+            startStaking(rewardsPools[i]);
```

Comment:
Wont this method fail, if we have at least one campaign already started?

Reply:
It would probably, my experience shows that we use this method only on the first deployment of the factory. Every other time we deploy one campaign at a time.

Reply:
I think that we don't need it if it is used for just one instance.